internal/mode/static/state/dataplane/configuration.go

```diff
@@ -559,3 +561,34 @@ func generateSSLKeyPairID(secret types.NamespacedName) SSLKeyPairID {
 func generateCertBundleID(configMap types.NamespacedName) CertBundleID {
 	return CertBundleID(fmt.Sprintf("cert_bundle_%s_%s", configMap.Namespace, configMap.Name))
 }
+
+// buildTelemetry generates the Otel configuration.
+func buildTelemetry(g *graph.Graph) Telemetry {
+	if g.NginxProxy != nil && g.NginxProxy.Spec.Telemetry != nil && g.NginxProxy.Spec.Telemetry.Exporter != nil {
```

Comment:
style:

if majority of the code is inside an if, it'd be more readable if we move that code out of it, so there is no need to indent it further

```
if g.NginxProxy == nil || g.NginxProxy.Spec.Telemetry == nil || g.NginxProxy.Spec.Telemetry.Exporter == nil {
  // return early
}

// most of the code
```

Related -- https://100go.co/#unnecessary-nested-code-2